drake_ros/core/camera_info_system.cc

```diff
@@ -0,0 +1,84 @@
+#include "drake_ros/core/camera_info_system.h"
+
+#include <rclcpp/time.hpp>
+
+using drake_ros::core::CameraInfoSystem;
+using drake_ros::core::RosPublisherSystem;
+
+CameraInfoSystem::CameraInfoSystem() : camera_info(1, 1, 1, 1, 0.5, 0.5) {
+  DeclareAbstractOutputPort("CameraInfoSystem",
+                            &CameraInfoSystem::CalcCameraInfo);
+}
+
+CameraInfoSystem::~CameraInfoSystem() {}
+
+void CameraInfoSystem::CalcCameraInfo(
+    const drake::systems::Context<double>& context,
+    sensor_msgs::msg::CameraInfo* output_value) const {
+  rclcpp::Time now{0, 0, RCL_ROS_TIME};
+  now += rclcpp::Duration::from_seconds(context.get_time());
+  output_value->header.stamp = now;
+  output_value->header.frame_id = "CartPole/camera_optical";
+
+  output_value->height = this->camera_info.height();
+  output_value->width = this->camera_info.width();
+  output_value->distortion_model = "plumb_bob";
+
+  output_value->d.push_back(0);
+  output_value->d.push_back(0);
+  output_value->d.push_back(0);
+  output_value->d.push_back(0);
+  output_value->d.push_back(0);
```

Comment:
Hmm this is weird that we're not using the distortion model parameters. Probably drake doesn't support distortion, so maybe this is fine : https://drake.mit.edu/doxygen_cxx/classdrake_1_1systems_1_1sensors_1_1_camera_info.html

Reply:
I'd be nice to add a comment saying the distortion isn't supported.